### PR TITLE
Replace asserts

### PR DESCRIPTION
**Pull request by maringuu**
Preliminary work for my incoming rework of the tests.

### CODE REVIEW

File: src/plugins/analysis/string_evaluation/test/test_string_eval.py [hunk 1]
```diff
@@ -24,17 +24,19 @@ class TestStringEval(unittest.TestCase):
...
         input_data = ['this string is useful', 'garbage n$%Schtz', '/an/interesting/directory']
         result = eval_strings(input_data)
-        self.assertEqual(
-            result, ['/an/interesting/directory', 'this string is useful', 'garbage n$%Schtz'], 'Fehler beim bewerten'
-        )
+        assert result == [
+            '/an/interesting/directory',
+            'this string is useful',
+            'garbage n$%Schtz',
+        ], 'Fehler beim bewerten'
```
Comment (maringuu) [inline src/plugins/analysis/string_evaluation/test/test_string_eval.py:31]: Err I don't think we should have german comments in our code.
I'll fix this in a future PR.

File: src/test/unit/helperFunctions/test_uid.py [hunk 2]
```diff
@@ -9,16 +9,16 @@ class TestHelperFunctionsUID(unittest.TestCase):
...
 
     def test_is_uid_list(self):
-        self.assertFalse(is_list_of_uids('blah'))
-        self.assertFalse(is_list_of_uids(['foobar']))
-        self.assertFalse(is_list_of_uids([]))
-        self.assertTrue([self.test_uid], 'uid list not recognized')
+        assert not is_list_of_uids('blah')
+        assert not is_list_of_uids(['foobar'])
+        assert not is_list_of_uids([])
+        assert [self.test_uid], 'uid list not recognized'
```
Comment (jstucke) [inline src/test/unit/helperFunctions/test_uid.py:24]: I'm pretty sure this was wrong all the time :rofl: 
```suggestion
        assert is_list_of_uids([self.test_uid]), 'uid list not recognized'
```

File: src/test/unit/helperFunctions/test_yara_binary_search.py [hunk 3]
```diff
@@ -64,24 +64,24 @@ def test_get_binary_search_yara_error(self, _):
     def test_eliminate_duplicates(self):
         test_dict = {1: [1, 2, 3, 3], 2: [1, 1, 2, 3]}
         self.yara_binary_scanner._eliminate_duplicates(test_dict)
-        self.assertEqual(test_dict, {1: [1, 2, 3], 2: [1, 2, 3]})
+        assert test_dict, {1: [1, 2, 3], 2: [1, 2 == 3]}
```
Comment (jstucke) [inline src/test/unit/helperFunctions/test_yara_binary_search.py:67]: .

File: src/test/unit/scheduler/test_compare.py [hunk 4]
```diff
@@ -72,22 +72,17 @@ def test_compare_single_run(self):
...
-            self.compare_scheduler._comparison_should_start('a', True, compares_done),
-            'redo is true so result should be true',
-        )
-        self.assertFalse(
-            self.compare_scheduler._comparison_should_start('a', False, compares_done),
-            'already done and redo no -> should be false',
-        )
+        assert self.compare_scheduler._comparison_should_start(
+            'b', False, compares_done
+        ), 'none existing should always be done'
```
Comment (jstucke) [inline src/test/unit/scheduler/test_compare.py:82]: I'm not even sure what the message should tell us ... and it should probably be "non-existing"

**Review by jstucke — CHANGES_REQUESTED** [review]
Not all tests were translated correctly into the new format

File: src/plugins/analysis/string_evaluation/test/test_string_eval.py [hunk 5]
```diff
@@ -24,17 +24,19 @@ class TestStringEval(unittest.TestCase):
...
 
     def test_add_mail_adress_score(self):
         input_data_true = 'here is my@mail.com'
         input_data_false = 'here is nothing'
         result_true = _add_mail_adress_score(input_data_true, 0)
         result_false = _add_mail_adress_score(input_data_false, 0)
-        self.assertEqual(result_true > 0, True, 'Mail false-negative')
-        self.assertEqual(result_false > 0, False, 'Mail false-positive')
+        assert result_true > 0, 'Mail false-negative'
+        assert result_false == 0, 'Mail false-positive'
```
[the remark this answers is not in the document]
Comment (maringuu) [reply src/plugins/analysis/string_evaluation/test/test_string_eval.py:39]: Yeah out of scope. Lets to such refactoring after #895 is finished.

**Comment by maringuu** [comment]
A addressed all your comments since the fixes were so leightweight.

**Pull request merged**